modules/configuration/src/main/java/org/apache/ignite/internal/configuration/asm/ConfigurationAsmGenerator.java

```diff
@@ -1238,6 +1238,8 @@ private void addNodeConstructMethod(
             FieldDefinition fieldDef = fieldDefs.get(fieldName);
             
             if (isPolymorphicId(schemaField)) {
+                makeSureChangePolymorphicTypeIdMethodIsDefined(changePolymorphicTypeIdMtd, schemaField);
```

Comment:
This is not the right place to check, it needs to be added on line 395.

Reply:
It is actually not a validation (which is made in `ConfigurationRegistry`), it is an assertion. Its function is to guard from a programming error, so it makes sense to put it as close as possible to the place in code where an NPE could be thrown.

Reply:
The assertion is that `changePolymorphicTypeIdMtd` is not `null` when we are going to use it. This variable is not yet in scope on line 395. It appears later.

Even if I create a validation on line 395, `changePolymorphicTypeIdMtd` usage is so far away that the assertion will still be needed. And the validation is actually made in other class (`ConfigurationRegistry`).

Do you want me to create a validation close to line 395 and remove the assertion?